src/steamcompmgr.cpp

```diff
@@ -6866,7 +6894,18 @@ steamcompmgr_main(int argc, char **argv)
 		{
 			gamescope_xwayland_server_t *server = NULL;
 			for (size_t i = 0; (server = wlserver_get_xwayland_server(i)); i++)
+			{
 				handle_done_commits_xwayland(server->ctx.get());
+
+				// When we have observed both a complete commit and a VBlank, we should request a new frame.
+				if (vblank)
```

Comment:
It is legal to get frame callbacks with no surface commits? I assumed that it is not possible, but if that is the case, this code would be problematic.

Reply:
I think without any commits we should be sending every frame if we are going to go with this approach

Reply:
I suppose a fix would be to check in latch_frame_callback if there are any pending commits. If not, mark commitDone instantly?

Reply:
Yeah, sounds good.

Reply:
`wl_surface.frame` just adds a pending callback. `commit` makes the pending state (incl. callbacks) current.

Reply:
Ok, I think this PR does the right thing then.

Reply:
In that case, Mutter and Weston are weird, because even with no commit in flight, they will trigger the frame callback on focus change, but only once.